apps/cowswap-frontend/src/modules/trade/containers/TradeWidget/index.tsx

```diff
@@ -14,7 +17,21 @@ export function TradeWidget(props: TradeWidgetProps) {
     tradeQuoteStateOverride,
     enableSmartSlippage,
   } = params
-  const modals = TradeWidgetModals({confirmModal, genericModal, selectTokenWidget: slots.selectTokenWidget})
+  const modals = TradeWidgetModals({ confirmModal, genericModal, selectTokenWidget: slots.selectTokenWidget })
+  const { isVisible } = useLimitOrdersPromoBanner()
+
+  // Inject the banner into the slots and use it as lockScreen when visible
+  const slotsWithBanner = {
+    ...slots,
+    topContent: (
+      <>
+        <LimitOrdersPromoBannerWrapper />
+        {slots.topContent}
+      </>
+    ),
+    // Only use banner as lockScreen when it's visible, otherwise use original lockScreen
+    lockScreen: isVisible ? <LimitOrdersPromoBannerWrapper /> : slots.lockScreen,
```

Comment:
I meant you can do:
```
<LimitOrdersPromoBannerWrapper>{slots.lockScreen}</LimitOrdersPromoBannerWrapper>
```

And inside of `LimitOrdersPromoBannerWrapper`:

```
function LimitOrdersPromoBannerWrapper({children}) {
const { isVisible } = useLimitOrdersPromoBanner()

if (!isVisible) return children
}
```

Anyway, since we don't have too much time, could you add TODO for that please? Not necessary to fix right now